packages/snaps-controllers/src/services/AbstractExecutionService.ts

```diff
@@ -55,6 +57,9 @@ export type Job<WorkerType> = {
   worker: WorkerType;
 };
 
+export type TerminateJobArgs<WorkerType> = Partial<Job<WorkerType>> &
```

Comment:
why the change in type for the terminate job args?

Reply:
I had to make the arguments partial to support termination of partially initialized executors, e.g. with the iframe executor the iframe has already been created in the DOM and we need to pass the ID. For most other executors this is not a problem.

https://github.com/MetaMask/snaps/pull/2348/files#diff-12564d49628a6405cf38ef5fdb16edd171449cb6f1f32301cdc068b94ae46cc2R252-R253